src/commcare_cloud/ansible/letsencrypt_cert.yml

```diff
@@ -41,11 +41,11 @@
 
     - name: Run Certbot command when there is a www host
       command: "certbot certonly --webroot -w /var/www/letsencrypt/ -m {{ root_email }} --agree-tos -n -d {{SITE_HOST}} -d {{ NO_WWW_SITE_HOST }}{% if ALTERNATE_HOSTS %} -d {{ ALTERNATE_HOSTS | join(' -d ') }}{% endif %} --expand"
-      when: NO_WWW_SITE_HOST|bool
+      when: NO_WWW_SITE_HOST is defined and NO_WWW_SITE_HOST
```

Comment:
You can also do `when: NO_WWW_SITE_HOST | default(False)`. I sometimes see this `|bool` thing in our ansible codebase and I think as you've discovered it's pretty misguided and almost never does the thing people put it there thinking it would do. `| default(False)` is pretty explicit and predictable. The expression `X | default(Y)` takes on the default value (Y) if and only if X is not defined, otherwise it takes on the value X.

Reply:
Nice, just tested that out and confirmed that it does what I want.  Switching to that.  This business where _coercing_ to a boolean yields a different result than _evaluating_ as a boolean has got me all flustered